### PR TITLE
Complete API for job groups

### PR DESCRIPTION
**Pull request by Martchus**
Required for further development of parent groups

### CODE REVIEW

File: assets/stylesheets/openqa.scss [hunk 1]
```diff
@@ -481,6 +481,11 @@ a.restart-link {
     margin-top: 20px;
 }
 
+/* admin job groups */
+#new_group_creating {
```
Comment (coolo) [inline assets/stylesheets/openqa.scss]: just make it class hidden and avoid the additional css rule

File: assets/javascripts/admin_groups.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+function createGroup(form) {
+    if(!form.name.value.length) {
+        window.alert('Group name mustn\'t be empty!');
```
Comment (coolo) [inline assets/javascripts/admin_groups.js]: alert is for debugging, jquery and bootstrap have so much more powerful for form validation

File: assets/javascripts/admin_groups.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    $('#new_group_creating').show();
+
+    var editorForm = $(form);
+    $.ajax({
+        url: editorForm.data('post-url'),
+        method: 'POST',
+        data: editorForm.serialize(),
+        success: function(response) {
+            window.location.pathname = 'admin/job_templates/' + response.id;
```
Comment (coolo) [inline assets/javascripts/admin_groups.js]: can't your response bring the URL? I don't want URLs hard coded in js.

File: lib/OpenQA/WebAPI.pm [hunk 4]
```diff
@@ -325,8 +324,8 @@ sub startup {
     # job_set_waiting, job_set_continue
     my $command_r = $job_r->route('/set_:command', command => [qw(waiting running)]);
     $command_r->post('/')->name('apiv1_set_command')->to('job#set_command');
-    $job_r->post('/restart')->name('apiv1_restart')->to('job#restart');     # job_restart
-    $job_r->post('/cancel')->name('apiv1_cancel')->to('job#cancel');        # job_cancel
+    $job_r->post('/restart')->name('apiv1_restart')->to('job#restart');            # job_restart
```
Comment (coolo) [inline lib/OpenQA/WebAPI.pm]: just remove these nonsense comments - they were referencing to the old (one to rule them all) Scheduler.pm. These comments change whitespace with every change and don't add any value.

File: lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm [hunk 5]
```diff
@@ -51,30 +51,51 @@ sub list {
...
 
 sub delete {
-    # TODO
+    my ($self) = @_;
+
+    my $group_id = $self->param('group_id');
+    my $group    = $self->db->resultset('JobGroups')->find($group_id);
+    return $self->render(json => {error => "Job group $group_id does not exist"}, status => 400) unless $group;
+
+    my $res = $group->delete;
```
Comment (coolo) [inline lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm]: What does with the jobs? please make sure it's covered in test cases

Comment (coolo) [inline lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm]: I can't think of any use case of deleting a job group and leaving the jobs. So only deleting empty groups sounds good, but then we need an API call to list jobs in a given group. I'm not sure we have yet.

**Comment by okurz** [comment]
also, doesn't seem to refresh the results when any value value in the pull down menue is selected

**Comment by coolo** [comment]
wrong PR commented ?

**Comment by okurz** [comment]
ah, *this* is where my comment ended up :-D Thank you for noticing

**Comment by okurz** [comment]
regarding *this* PR: LGTM, the next obvious parts that are missing is deleting a job group to complete the CRUD cycle :-)
Also, I assume the description is not yet used and therefore does not show up yet.

**Pull request merged**